### PR TITLE
Add references to OAuth 2.0 Security Best Practices for Implicit and

### LINKED ISSUE #23480: Docs: securing_apps/topics/oidc/supported-grant-types.adoc
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

docs

### Describe the bug

File: securing_apps/topics/oidc/supported-grant-types.adoc

The documentation pertaining to Implicit flow and ROPC could do better to link to current OAuth 2.0 best practice and indicate those flows are not recommended for use and are removed from OAuth 2.1 spec (in draft now).

Additionally, the ROPC wording could be changed to point to device auth or authorization code as preferred alternatives, and also do a better job of indicating the security implications of using ROPC within a deployment of keycloak.

### Version

22.0.3

### Expected behavior

See description

### Actual behavior

Updated docs reflecting the updates I mentioned in the description.

### How to Reproduce?

Build the docs

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jsorah**
ROPC flow, reformat Device Auth section.

Closes keycloak/keycloak#23480

### CODE REVIEW

**Review by mposolda — DISMISSED** [review]
@jsorah Thanks!
@andymunro Is this PR ok to you?

File: docs/documentation/securing_apps/topics/oidc/supported-grant-types.adoc [hunk 1]
```diff
@@ -28,10 +28,15 @@ browser history. You can somewhat mitigate this problem by using short expiratio
 
 For more details, see the https://openid.net/specs/openid-connect-core-1_0.html#ImplicitFlowAuth[Implicit Flow] in the OpenID Connect specification.
 
+Per current https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-23#name-implicit-grant[OAuth 2.0 Security Best Current Practice] this flow should not be used.
```
Comment (andymunro) [inline docs/documentation/securing_apps/topics/oidc/supported-grant-types.adoc:31]:
```suggestion
Per current https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-23#name-implicit-grant[OAuth 2.0 Security Best Current Practice], this flow should not be used.
```

File: docs/documentation/securing_apps/topics/oidc/supported-grant-types.adoc [hunk 2]
```diff
@@ -28,10 +28,15 @@ browser history. You can somewhat mitigate this problem by using short expiratio
...
+This flow is removed from the future https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-09[OAuth 2.1 specification].
+
 [[_resource_owner_password_credentials_flow]]
 ==== Resource Owner Password Credentials
 
-Resource Owner Password Credentials, referred to as Direct Grant in {project_name}, allows exchanging user credentials for tokens. Using this flow is not recommended unlesss it is essential. Examples where this flow could be useful are legacy applications and command-line interfaces.
+Resource Owner Password Credentials, referred to as Direct Grant in {project_name}, allows exchanging user credentials for tokens.
+Per current https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-23#name-resource-owner-password-cre[OAuth 2.0 Security Best Practices]
```
Comment (andymunro) [inline docs/documentation/securing_apps/topics/oidc/supported-grant-types.adoc:38]:
```suggestion
Per current https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-23#name-resource-owner-password-cre[OAuth 2.0 Security Best Practices],
```

File: docs/documentation/securing_apps/topics/oidc/supported-grant-types.adoc [hunk 3]
```diff
@@ -78,12 +90,13 @@ For more details, see the https://datatracker.ietf.org/doc/html/rfc6749#section-
...
 
 Device Authorization Grant is used by clients running on internet-connected devices that have limited input capabilities or lack a suitable browser.
-. The application requests {project_name} a device code and a user code. 
-. {project_name} creates a device code and a user code. 
-. {project_name} returns a response including the device code and the user code to the application. 
-. The application provides the user with the user code and the verification URI. The user accesses a verification URI to be authenticated by using another browser. 
-. The application repeatedly polls {project_name} until {project_name} completes the user authorization. 
-. If user authentication is complete, the application obtains the device code. 
+
+. The application requests {project_name} a device code and a user code.
```
Comment (andymunro) [inline docs/documentation/securing_apps/topics/oidc/supported-grant-types.adoc:94]:
```suggestion
. The application requests  that {project_name} provide a device code and a user code.
```

**Review by andymunro — CHANGES_REQUESTED** [review]
Thanks, @mposolda. Just  few changes.

**Comment by jsorah** [comment]
@andymunro @mposolda  - accepted changes from code review and squashed commits.

**Comment by mposolda** [comment]
@jsorah @andymunro Thanks for the review and follow-up changes!

**Review by mposolda — APPROVED** [review]

**Pull request merged**